test/groovy/CloudFoundryDeployTest.groovy

```diff
@@ -1069,4 +1070,81 @@ class CloudFoundryDeployTest extends BasePiperTest {
 
     }
 
+    @Test
+    void testMtaExtensionCredentials() {
+        fileExistsRule.existingFiles.addAll(
+            'mtaext.mtaext',
+        )
+        credentialsRule.withCredentials("mtaExtCredTest","token")
+
+        helper.registerAllowedMethod('readFile', [String], {file ->
+            if (file == 'mtaext.mtaext') {
+                return '<%= testCred %>'
+            }
+            return ''
+        })
+
+        stepRule.step.cloudFoundryDeploy([
+            script: nullScript,
+            juStabUtils: utils,
+            jenkinsUtilsStub: new JenkinsUtilsMock(),
+            cfOrg: 'testOrg',
+            cfSpace: 'testSpace',
+            cfCredentialsId: 'test_cfCredentialsId',
+            deployTool: 'mtaDeployPlugin',
+            mtaPath: 'target/test.mtar',
+            mtaExtensionDescriptor: "mtaext.mtaext",
+            mtaExtensionCredentials: [
+                testCred: 'mtaExtCredTest'
+            ]
+        ])
+
+        assertThat(shellRule.shell, hasItem(containsString('cp mtaext.mtaext mtaext.mtaext.original')))
+        assertThat(shellRule.shell, hasItem(containsString('mv --force mtaext.mtaext.original mtaext.mtaext')))
+        assertThat(writeFileRule.files['mtaext.mtaext'], containsString('token'))
```

Comment:
You could improve this assertion by not just testing for contains. I think you should have a little more complex extension file and that only the place holder is replaced but not the rest of the file.

Reply:
Yes that's right I improved the test as suggested.